src/Workspaces/VisualBasic/Portable/CodeGeneration/EventGenerator.vb

```diff
@@ -64,7 +64,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGeneration
                                                  options As CodeGenerationOptions) As DeclarationStatementSyntax
             Dim reusableSyntax = GetReuseableSyntaxNodeForSymbol(Of DeclarationStatementSyntax)([event], options)
             If reusableSyntax IsNot Nothing Then
-                Return reusableSyntax
+                Return DirectCast(reusableSyntax.GetBlockFromBegin(), DeclarationStatementSyntax)
```

Comment:
It might be a better idea to change `GetBlockFromBegin` return type to `DeclarationStatementSyntax` instead of doing the cast.

Reply:
In the `GetBlockFromBegin` method (linked [here](https://github.com/dotnet/roslyn/blob/082df2aaf6a73277fc27f9d7a5a8d18ee45d7f8b/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/VisualBasic/Extensions/SyntaxNodeExtensions.vb#L1249)), we either return the current node, which is guaranteed to be a `DeclarationStatementSyntax`, or we return the parent, which is also a class derived from `DeclarationStatementSyntax` in all cases except when the parent is a `VariableDeclaratorSyntax`. We shouldn't ever have a `VariableDeclaratorSyntax` as our parent, because this only changes on the generation of methods, properties, and events, so I think this cast is safe. 

I think we could change the other casts to also be `DeclarationStatementSyntax`, since it is derived from `StatementSyntax`. It should work the same, as every `StatementSyntax` returned from `GetBlockFromBegin` is also a `DeclarationStatementSyntax`. 

We could alternatively write some overload/equivalent method of `GetBlockFromBegin` that removes the separate case for the parent being a `VariableDeclaratorSyntax`. Then, we could both require a `DeclarationStatementSyntax` as the input and also guarantee the return of a `DeclarationStatementSyntax`.

Reply:
I'll defer to @CyrusNajmabadi on what to do (if any action is required)

Reply:
yes.  i think this should become strongly typed to try to return a DeclarationStatementSyntax.  Try making that change and see if anything breaks with a cast-exception.  if so, let us know and we can decide what to do at that point.  if not, then let's definitely go that route.  thanks!